### ISSUE TITLE
wasm-snip stage of build fails

### ISSUE THREAD

**Issue by Kagami** [fitzgen/source-map-mappings#16]
Hey.

I wasy trying to build package using instructions from CONTRIBUTING.md, but it fails with the following error:

```
$ ./build.py -o mappings.wasm                                                              
['cargo', 'build', '--release', '--target', 'wasm32-unknown-unknown']
    Finished release [optimized + debuginfo] target(s) in 0.0 secs
['wasm-gc', './target/wasm32-unknown-unknown/release/source_map_mappings_wasm_api.wasm', './target/wasm32-unknown-unknown/release/source_map_mappings_wasm_api.gc.wasm']
['wasm-nm', '-p', '-j', './target/wasm32-unknown-unknown/release/source_map_mappings_wasm_api.gc.wasm']
['wasm-snip', '-o', './target/wasm32-unknown-unknown/release/source_map_mappings_wasm_api.gc.snip.wasm', './target/wasm32-unknown-unknown/release/source_map_mappings_wasm_api.gc.wasm']
error: The following required arguments were not provided:
    <function>...

USAGE:
    wasm-snip <input> <function>... --output <output>

For more information try --help
```

With `--no-wasm-snip` it works. Seems like wasm-snip returns mangled symbols while build script expects them to be demangled:

```
$ wasm-nm -p -j target/wasm32-unknown-unknown/release/source_map_mappings_wasm_api.gc.wasm | head -10
get_last_error
_ZN49_$LT$alloc..raw_vec..RawVec$LT$T$C$$u20$A$GT$$GT$6double17hf04e4e57ced8d322E
_ZN4core3fmt5Write9write_fmt17h6bc7e1af287ba745E
_ZN4core3ptr13drop_in_place17hcffacfeea73e0053E
_ZN53_$LT$$RF$$u27$a$u20$T$u20$as$u20$core..fmt..Debug$GT$3fmt17hf4ab534754c36a85E
_ZN3std2io5error5Error3new17hdc50c5866a8ba7efE
_ZN33_$LT$alloc..arc..Arc$LT$T$GT$$GT$9drop_slow17hd540ef190adfe368E
_ZN33_$LT$alloc..vec..Vec$LT$T$GT$$GT$17extend_from_slice17had06c280e2d0e5dcE
_ZN4core6result13unwrap_failed17h430e08947ed8d5e6E
_ZN4core6result13unwrap_failed17h6267ff1f2864ef32E
```

Any ideas?

**Comment by fitzgen** [comment]
What OS are you on? I feel like I've seen differences between OSX and linux around whether the `"name"` section contains mangled or demangled names.

We should be able to pipe through `rustfilt` to normalize everything to demangled names, if that helps.

I'm a little confused though, because both `wasm-snip` and `wasm-nm` use the same `"name"` section, and so it shouldn't matter whether or not the symbol names are mangled or not, as long as they agree...

**Comment by Kagami** [comment]
I'm on Gentoo Linux x86-64.

>I'm a little confused though, because both wasm-snip and wasm-nm use the same "name" section

The problem is in build.py - it tries to find demangled symbols:
https://github.com/fitzgen/source-map-mappings/blob/master/source-map-mappings-wasm-api/build.py#L96-L117

**Comment by fitzgen** [comment]
The easiest solution is probably to make the regexps work with either mangled or demangled symbols.

**Comment by fitzgen** [comment]
Fix over in #17
